regions/core/pixcoord.py

```diff
@@ -33,9 +33,24 @@ class PixCoord(object):
     """
 
     def __init__(self, x, y):
+
         self.x = self._standardize_coordinate(x)
         self.y = self._standardize_coordinate(y)
 
+        for a, b in ((x, y), (y, x)):
```

Comment:
@sushobhana - Can we avoid these ~ 10 complex lines by re-using some utility function from Astropy or Numpy?
E.g. would this help? https://docs.scipy.org/doc/numpy/reference/generated/numpy.broadcast_arrays.html
I had a quick look at SkyCoord but couldn't find the code where this input array broadcasting is done there.

Reply:
Another option might be to always go to arrays in the init, and only handle the special case of scalars at the end of init. See https://mail.python.org/pipermail/numpy-discussion/2017-December/077512.html

Note that my previous comment on inefficiency of creating Numpy arrays was only because you just did that to get the shape, and then later again; and overall it probably wasn't the most important point / best comment.

I really like the changes you've made here, with the behaviour demonstrated via the tests, so let's try to simplify the init if possible, but in any case merge this in soon.

Reply:
@cdeil great!!
The `numpy.broadcast_arrays` seems to work (tried in the repl), this would simplify the problem.
it is also used for  SkyCoord. The code is present in astropy/coordinates/representation.py file. (line:171)